migrator/clone/postgres/db_clone_manager_impl.go

```diff
@@ -38,9 +39,55 @@ func New(forceVersion string, adminConfig *postgres.Config, sourceMap map[string
 	}
 }
 
+func (d *dbCloneManagerImpl) getVersion() (*migrations.MigrationVersion, error) {
+	ctx := sac.WithAllAccess(context.Background())
+
+	gc := migGorm.GetConfig()
+	db, err := gc.ConnectWithRetries(d.adminConfig.ConnConfig.Database)
+	if err != nil {
+		return nil, err
+	}
+	defer migGorm.Close(db)
+
+	return migVer.ReadVersionGormDB(ctx, db)
+}
+
+func (d *dbCloneManagerImpl) ensureVersionCompatible() error {
+	ver, err := d.getVersion()
+	if err != nil {
+		return err
+	}
+	log.Infof("db is of version %v", ver)
+
+	if d.versionExists(ver) {
+		// If the database version is newer than the software version make the user explicitly state they want to rollback.
+		// TODO:  Do we still want to do this `forceRollbackVersion` check?
```

Comment:
I don't think we want the force rollback check unless we integrate it into the operator. Currently, we're working around this quite a bit in the operator code which does not handle this use case